app/controllers/users/registrations_controller.rb

```diff
@@ -49,23 +40,17 @@ def email_domain
     @email.split("@")[1]
   end
 
-  def handle_matching_schools
-    unclaimed_schools = @schools.filter { |school| school.induction_coordinator_profiles.none? }
-
-    if unclaimed_schools.any?
-      redirect_to controller: "users/registrations", action: :confirm_school, school_ids: unclaimed_schools, email: @email
-    else
-      redirect_to root_path, alert: "Someone from your school has already signed up"
-    end
+  def render_start_registration
+    flash.now[:alert] = "Your details did not match any schools."
+    render :start_registration
   end
 
   def validate_creation_parameters
-    @school = School.find(params[:user][:school_id])
+    @school = School.find_by(id: params[:user][:school_id])
```

Comment:
I thought those are identical. Are they not?

Reply:
find_by won't cause an exception if there is no record. It will just return nil instead.